boolean_parser/mixins/sqla.py

```diff
@@ -166,7 +194,35 @@ def _filter_one(self, model, field=None, condition=None):
         elif self.operator == '>=':
             condition = lower_field.__ge__(lower_value)
         elif self.operator == '!=':
-            condition = lower_field.__ne__(lower_value)
+            if isinstance(field.type, sqltypes.TEXT) or \
+                isinstance(field.type, sqltypes.VARCHAR) or \
+                isinstance(field.type, sqltypes.String):
+                field = getattr(model, self.name)
+                value = self.value
+                if value.lower() == 'null':
+                    condition = field.isnot(None)
+                else:
+                    condition = lower_field.__ne__(lower_value)
+            elif isinstance(field.type, sqltypes.BOOLEAN) or \
+                isinstance(field.type, sqltypes.Boolean):
+                field = getattr(model, self.name)
+                value = self.value
+                if value.lower() == 'null':
+                    condition = field.isnot(None)
+                else:
+                    condition = lower_field.__ne__(lower_value)
+            elif isinstance(field.type, sqltypes.DATE) or \
+                isinstance(field.type, sqltypes.DATETIME) or \
+                isinstance(field.type, sqltypes.Date) or \
+                isinstance(field.type, sqltypes.DateTime):
+                field = getattr(model, self.name)
+                value = self.value
+                if value.lower() == 'null':
+                    condition = field.isnot(None)
+                else:
+                    condition = lower_field.__ne__(lower_value)
+            else:
+                condition = lower_field.__ne__(lower_value)
         elif self.operator == '=':
```

Comment:
The code under this operator `=` block is also now very similar to the code for the `==` and `!=` operators.  I'm wondering if there's a way to generalize this along with those code blocks and simplify it a bit?  The `ilike` makes it a bit different but just a thought.

Reply:
see above.